services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/utils/SubnetManagementUtil.java

```diff
@@ -206,4 +206,42 @@ private static long getIpNum(String ipAddress) {
     private static boolean isInner(long userIp, long begin, long end) {
         return (userIp >= begin) && (userIp <= end);
     }
+
+    public static boolean IsCidrWithin (String cidr1, String cidr2) {
+        if (cidr1 == null || cidr2 == null || cidr1.length() == 0 || cidr2.length() == 0) {
+            return true;
```

Comment:
Consider to "return false" or throw exception. 

If we really want to return true, there will be more complicated scenarios to handle, for example, cidr1=<Valid cidr> and cidr2=NULL